File: nte-aisdk-release-2.0/nte-aisdk-release-2.0/src/nte_aisdk/vector_store.py
from elasticsearch import Elasticsearch


class VectorStore:
    """VectorStore is a base class to extend the functionalitiy for ElasticSearch client.

    Args:
        host: The host of the ElasticSearch instance.
        basic_auth: The tuple of username and password of the ElasticSearch user.

    """

    def __init__(
            self,
            host: str,
            basic_auth: str | tuple[str, str],
        ):
        self.es = Elasticsearch(
            host,
            basic_auth=basic_auth
        )

File: nte-ai-sdk-node-env-d0/nte-ai-sdk-node-env-d0/src/vector-store.ts
import { Client } from '@elastic/elasticsearch';

export interface VectorStoreOptions {
  /**
   * The host of the Elasticsearch instance.
   */
  host: string;

  /**
   * The basic auth credentials for the Elasticsearch instance.
   */
  basicAuth: {
    username: string;
    password: string;
  };
}

export class VectorStore {
  es: Client;

  constructor(options: VectorStoreOptions) {
    this.es = new Client({
      node: options.host,
      auth: {
        username: options.basicAuth.username,
        password: options.basicAuth.password,
      }
    })
  }
}